### ISSUE TITLE
false positive for Lint/ToEnumArguments when using sorbet

### ISSUE THREAD

**Issue by dduugg** [rubocop/rubocop#11591]
## Expected behavior

The "good" [example](https://docs.rubocop.org/rubocop/cops_lint.html#linttoenumarguments) for `Lint/ToEnumArguments` looks like this:

```ruby
def foo(x, y = 1)
  to_enum(__callee__, x, y)
end
```

However, since `__callee__` and `__method__` can return `nil`, these are not viable in a type-checked ruby context, Sorbet in particular:

https://sorbet.run/#%23%20frozen_string_literal%3A%20true%0A%0Aextend%20T%3A%3ASig%0A%0Adef%20foo%28x%2C%20y%20%3D%201%29%0A%20%20to_enum%28__callee__%2C%20x%2C%20y%29%0Aend%0A%0Adef%20bar%28x%2C%20y%20%3D%201%29%0A%20%20to_enum%28T.must%28__callee__%29%2C%20x%2C%20y%29%0Aend

Instead, the `__callee__` invocation has to be [asserted](https://sorbet.org/docs/type-assertions#tmust) as non-`nil`, by wrapping in `T.must`:

```ruby
def for(x, y = 1)
  to_enum(T.must(__callee__), x, y)
end
```

I believe that this formulation should be accepted by `rubocop`.

## Actual behavior

The last example above triggers a rubocop violation:

```
foo.rb:10:3: W: Lint/ToEnumArguments: Ensure you correctly provided all the arguments.
  to_enum(T.must(__callee__), x, y)
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

## Steps to reproduce the problem

Invoke `rubocop foo.rb --only Lint/ToEnumArguments`, where `foo.rb` contains the following:
```ruby
def bar(x, y = 1)
  to_enum(T.must(__callee__), x, y)
end
```

## RuboCop version
```
$ rubocop -V
1.45.1 (using Parser 3.2.0.0, rubocop-ast 1.24.1, running on ruby 3.1.1) [x86_64-darwin21]
```
